lib/services/server.go

```diff
@@ -428,3 +433,41 @@ func NodeHasMissedKeepAlives(s types.Server) bool {
 	serverExpiry := s.Expiry()
 	return serverExpiry.Before(time.Now().Add(apidefaults.ServerAnnounceTTL - (apidefaults.ServerKeepAliveTTL() * 2)))
 }
+
+// NewAWSNodeFromEC2Instance creates a Node resource from an EC2 Instance.
+// It has a pre-populated spec which contains info that is not available in the ec2.Instance object.
+func NewAWSNodeFromEC2Instance(instance ec2Types.Instance, awsCloudMetadata *types.AWSInfo) (types.Server, error) {
+	labels := libaws.TagsToLabels(instance.Tags)
+	if labels == nil {
+		labels = make(map[string]string)
```

Comment:
Is this mandatory? What would fail if we didn't do it?

Reply:
It might be the case that `instance.Tags` is an empty list and `labels` will be nil.
Then, `AddMetadataLabels` expects a non-nil map.
If we don't nil-check its value, it might panic here